### ISSUE TITLE
Use aerial.nvim as the symbols viewer (instead of symbols-outline.nvim) due to ability to use treesitter for symbols extraction

### ISSUE THREAD

**Issue by krishnakumarg1984** [AstroNvim/AstroNvim#377]
**Is your feature request related to a problem? Please describe.**
symbols-outline.nvim requires a properly configured LSP server & attached to the current buffer. 

- Installing an LSP is not always possible, especially when working on remote machines without admin access. 
- As a workaround for the lack of sudo access, it can be argued that userspace installs are possible (with brew, nvim-lspinstall, compile from source etc), there are several problems in doing this:
1. A lot of LSPs require nodejs as a dependency (which may not be available or needs to be built!)
2. Even if we get past that, some LSPs require a very modern runtime environment. For example, clangd requires a fairly recent version of `glibc`. In machines that still run CentOS 7 (which is still supported) such as in the HPCs I use, administrators of such national supercomputing facilties shall not update these core infrastructure packages.

We need a plugin that does not rely on LSP to provide a symbols-tree outline viewer. 

**Describe the solution you'd like**
Use a package that can leverage `treesitter` for providing symbols-tree viewer. A plugin that I have been using for the last 5 months is [aerial.nvim](https://github.com/stevearc/aerial.nvim), and I have been satisfied with it. It is actively maintained, and the author is responsive to address questions & GitHub issues. 

Quoting the relevant line of text from the project's README:
>First ensure you have [nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter) installed and configured for all languages you want to support. That's all! Aerial will automatically fetch symbols from treesitter.

On remote machines without admin access, there are no impediments to using treesitter, since there is built-in support in installing the relevant parsers in core `neovim` itself (the latest version of nvim can be used as an appimage downloaded to my `$HOME/bin`).

**Comment by mehalter** [comment]
Thanks for the recommendation @krishnakumarg1984 this is a great plugin!

**Comment by krishnakumarg1984** [comment]
Thank you for accepting the plugin suggestion. Just a couple of follow-on things:
- I notice a typo in a couple of places. `Aerial` is spelt without the `e` as `Arial` in the mappings.
- The companion plugin [stickybuf.nvim](https://github.com/stevearc/stickybuf.nvim) by the same author helps to prevent accidentally opening regular files in the tiny aerial split. Could you please have a look and consider it for inclusion if you deem it is useful?

**Comment by mehalter** [comment]
@krishnakumarg1984 thanks for the heads up on this stuff! I just made a PR (#401) to address the typos and also use the telescope fuzzy finder if it is available instead of `lsp_document_symbols`. Their picker is much nicer. I'll need to find some more time to look into `stickybuf.nvim`. Also I can help you with the lualine stuff in the coming days.

**Comment by krishnakumarg1984** [comment]
@mehalter Thank you for so quickly helping to address the queries!
- The document symbols mapping and the telescope picker works flawlessly after I pulled and tested your latest commits. 
- The lualine integration you suggested works for me

There are some last vestiges of the `symbols_outline` plugin in AstroNvim e.g. in `lua/core/defaults.lua` (line 27) and in `lua/user_example/init.lua` (line 31) which are used for enabling and disabling that plugin (and perhaps requiring corresponding changes to the relevant callback functions that do the actual enabling/disabling). Perhaps these should all be updated for `aerial.nvim`? 

Would it be greedy of me to ask for a solution of lualine integration with aerial lazy loading? Essentially, only if
- treesitter parser is installed
- an lsp server is attached for the current buffer 
we need to really load `aerial`. Otherwise `aerial.nvim` doesn't do anything at all, right? For example, if I open a filetype that has neither a treesitter parser installed, nor we have installed and connected to a valid LSP, why do we need to incur the penalty overheads of loading `aerial.nvim`? 

Finally, shall I make a separate issue for investigating [stickybuf.nvim](https://github.com/stevearc/stickybuf.nvim), since this issue has been closed, and might help with visibility of TBD things in the project?

A big thanks to you for developing and maintaining this awesome neovim configuration.
